### PR TITLE
fix(console): use correct logic to display refresh token switches

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

fix display logic for "Always issue refresh token" and "Refresh token TTL" according to OIDC configuration

## Testing

n/a

## Checklist

- [x] `.changeset`

### CODE REVIEW

File: packages/console/src/pages/ApplicationDetails/components/AdvancedSettings.tsx [hunk 1]
```diff
@@ -121,54 +121,65 @@ function AdvancedSettings({ applicationType, oidcConfig }: Props) {
...
-              placeholder="14"
-              // Confirm if we need a customized message here
-              error={errors.customClientMetadata?.refreshTokenTtlInDays?.message}
-            />
-          </FormField>
-        </>
-      )}
+      {/**
+        * Public clients (authentication method is none) are not allowed to disable refresh token
+        * rotation, so we don't show the option here.
```
Comment (charIeszhao) [inline packages/console/src/pages/ApplicationDetails/components/AdvancedSettings.tsx:126]: I think even for non-sender constraint public clients, we can still disable refresh token rotation. It is only by default that this option is enabled but it doesn't mean we can't turn it off.

File: .changeset/smooth-radios-wink.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+"@logto/console": patch
+---
+
+fix display logic for "Always issue refresh token" and "Refresh token TTL" according to OIDC configuration
```
Comment (charIeszhao) [inline .changeset/smooth-radios-wink.md:5]: It's nothing about "Always issue refresh token".
The change actually impacts "Rotate refresh token" toggle switch and "Refresh token TTL" field

Comment (gao-sun) [reply .changeset/smooth-radios-wink.md:5]: i was unconscious

File: packages/console/src/pages/ApplicationDetails/components/AdvancedSettings.tsx (hunk 1 above)
Comment (gao-sun) [reply packages/console/src/pages/ApplicationDetails/components/AdvancedSettings.tsx:126]: if we follow the default logic of oidc-provider, they should be able to turn off this

Comment (charIeszhao) [reply packages/console/src/pages/ApplicationDetails/components/AdvancedSettings.tsx]: Per offline discussion, the users should still be able to to turn it off even if it's SPA, although it is not recommended.

**Comment by charIeszhao** [comment]
After investigation, our current implementation is actually correct, we don't need to "fix" anything.

Conclusion:
* SPA apps are considered non-sender constrained clients
* When refresh token rotation is enabled, refresh tokens that issued to SPA clients will be rotated in every token exchange request, issuing a new access token as well as a new refresh token
* SPA apps can still disable refresh token rotation, although it is strongly recommended NOT to.
* SPA apps can adjust refresh token TTL. But if token rotation is enabled, the TTL will not be respected in token exchange request. In other words, the refresh token will still be rotated every time it is consumed to exchange an access token, regardless of its remaining TTL.

**Pull request closed without merging**